File: core/core-domain/src/main/java/com/coffee/gu/payment/PaymentRepository.java
package com.coffee.gu.payment;


import java.util.List;
import java.util.Optional;

public interface PaymentRepository {
    Optional<Payment> findByOrderKey(String orderKey);
    Payment save(Payment payment);
    Optional<Payment> findByIdWithLock(Long id);
    Optional<Payment> findByOrderKeyWithLock(String orderKey);
    List<Payment> getPendingPayments(int limit);
}

File: core/core-domain/src/main/kotlin/com/coffee/gu/payment/PaymentRepository.kt
package com.coffee.gu.payment

interface PaymentRepository {
    fun findByOrderKey(orderKey: String): Payment
    fun save(payment: Payment): Payment
    fun findByIdWithLock(id: Long): Payment
    fun findByOrderKeyWithLock(orderKey: String): Payment
    fun claimPendingPayments(limit: Int): List<Payment>
}
